### PR TITLE
Add Bigtable Emulator support

### LINKED ISSUE #2319: BIGTABLE_EMULATOR_HOST support
https://cloud.google.com/bigtable/docs/emulator

My attempt at running the V2 Bigtable client against a local emulator:

I figured out that I needed to add support for passing on `:this_channel_is_insecure` to the client sub
so I added the following code in the *Client classes
```ruby
if credentials.is_a?(GRPC::Core::ChannelCredentials)
  chan_creds = credentials
end
if credentials == :this_channel_is_insecure
  chan_creds = credentials
end
```
My attempt:
```ruby
require 'google/cloud/bigtable'
class LocalBigtableInstanceAdminClient < Google::Cloud::Bigtable::Admin::V2::BigtableInstanceAdminClient
  SERVICE_ADDRESS = 'localhost'
  DEFAULT_SERVICE_PORT = 8086
end

class LocalBigtableTableAdminClient < Google::Cloud::Bigtable::Admin::V2::BigtableTableAdminClient
  SERVICE_ADDRESS = 'localhost'
  DEFAULT_SERVICE_PORT = 8086
end
class LocalBigtableClient < Google::Cloud::Bigtable::V2::BigtableClient
  SERVICE_ADDRESS = 'localhost'
  DEFAULT_SERVICE_PORT = 8086
end

bigtable_instance_admin = LocalBigtableInstanceAdminClient.new(credentials: :this_channel_is_insecure)
bigtable_table_admin = LocalBigtableTableAdminClient.new(credentials: :this_channel_is_insecure)
bigtable = LocalBigtableClient.new(credentials: :this_channel_is_insecure)

bigtable_instance_admin.create_instance('project', 'test_instance', {}, {})
```
```
.rvm/gems/ruby-2.5.1/gems/google-gax-1.3.0/lib/google/gax/api_callable.rb:369:in `rescue in block in retryable': GaxError Exception occurred in retry method that was not classified as transient, caused by 12:unknown service google.bigtable.admin.v2.BigtableInstanceAdmin (Google::Gax::RetryError)
```

### PR DESCRIPTION
**Pull request by blowmage**
* Allow the emulator host to be provided in the `BIGTABLE_EMULATOR_HOST` environment variable, or the `emulator_host` constructor argument.
* Add `EMULATOR.md` guide to show how to configure and use the emulator.

[closes #2319]

### CODE REVIEW

**Comment by blowmage** [comment]
FYI @sduskis @jiren. (We don't have permissions to add @jiren as a reviewer.)

**Comment by blowmage** [comment]
My manual testing has made me believe that the emulator supports only some of the API endpoints. I think the `EMULATOR.md` guide should provide more information about what calls are supported, but I don't know for sure what is and isn't supported. I would like to get additional information in the guide if possible. Can someone inform us about this?

File: google-cloud-bigtable/lib/google/cloud/bigtable.rb [hunk 1]
```diff
@@ -69,18 +71,28 @@ module Bigtable
...
+          return Bigtable::Project.new(
+            Bigtable::Service.new(
+              project_id, :this_channel_is_insecure,
+              host: emulator_host, timeout: timeout,
+              client_config: client_config
+            )
+          )
+        end
 
+        credentials ||= default_credentials(scope: scope)
```
Comment (sduskis) [inline google-cloud-bigtable/lib/google/cloud/bigtable.rb]: credentials are not required in emulator mode, so emulator users should not fail for empty credentials.

Comment (blowmage) [reply google-cloud-bigtable/lib/google/cloud/bigtable.rb]: Correct, if it is using the emulator the correct object will be returned above, on line 86.

**Comment by blowmage** [comment]
@sduskis Does this look good? If so, can you approve?

**Pull request merged**